app/src/main/java/com/io/movio/adaptor/MoviesAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package com.io.movio.adaptor
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.io.movio.databinding.MovieRowBinding
+import com.io.movio.models.Movie
+
+class MoviesAdapter(private val movies: Array<Movie>,
+                    private val listener: ItemOnClickListener)
+    : RecyclerView.Adapter<MoviesAdapter.MovieViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder
+    = MovieViewHolder(MovieRowBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+
+    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
+
+        holder.apply {
```

Comment:
So, why do you have holder.apply{} but you keep using the holder inside the block?